### PR TITLE
Add simple timings from WalshyDev

### PR DESCRIPTION
**Pull request by md5sha256**
Adds a modified version of a patch written by @WalshyDev initially for Slimefun.

### CODE REVIEW

**Comment by WalshyDev** [comment]
The build is failing because it looks like one of the repos is dead, it has the wrong certificate issued to it. This is why I don't like using so many third-party repos *sigh*

**Comment by WalshyDev** [comment]
Looks like MyPet is open source and we can get this from jitpack - https://github.com/scomans/MyPet

**Comment by md5sha256** [comment]
Doesn't MP use NMS?

**Comment by WalshyDev** [comment]
It does but we only need the API module

**Comment by md5sha256** [comment]
Good point

**Comment by WalshyDev** [comment]
We should add a buildTimings method that compiles it into a string. Then add 2 new `printTimings` which accept a `CommandSender`

**Comment by md5sha256** [comment]
I wonder if it makes more sense to name methods `logTimings` for methods which print to the logger and reserve `printTimings` to sending to `CommandSender`

**Review by WalshyDev — DISMISSED** [review]
LGTM

**Comment by WalshyDev** [comment]
> I wonder if it makes more sense to name methods `logTimings` for methods which print to the logger and reserve `printTimings` to sending to `CommandSender`

I think it's time. It's still being printed

**Comment by md5sha256** [comment]
So, logTime and printTime?

File: dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    public void logTimings(boolean clearTimings) {
+        printTimings(clearTimings, logger::info);
+    }
+
+    private void printTimings(boolean clearTimings, @Nonnull Consumer<String> printer) {
+        printer.accept(buildTimings(clearTimings));
+    }
+
+    public void printTimings(boolean clearTimings, @Nonnull CommandSender sender) {
```
Comment (WalshyDev) [inline dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java:73]:
```suggestion
    private void logTimings(boolean clearTimings, @Nonnull Consumer<String> printer) {
        printer.accept(buildTimings(clearTimings));
    }

    public void logTimings(boolean clearTimings, @Nonnull CommandSender sender) {
```

File: dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        printTimings(clearTimings, logger::info);
+    }
+
+    private void logTimings(boolean clearTimings, @Nonnull Consumer<String> printer) {
+        printer.accept(buildTimings(clearTimings));
+    }
+
+    public void logTimings(boolean clearTimings, @Nonnull CommandSender sender) {
+        printTimings(clearTimings, sender::sendMessage);
+    }
```
Comment (WalshyDev) [inline dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java:75]:
```suggestion
    public void logTimings(boolean clearTimings, @Nonnull CommandSender sender) {
        logTimings(clearTimings, sender::sendMessage);
    }

    private void logTimings(boolean clearTimings, @Nonnull Consumer<String> printer) {
        printer.accept(buildTimings(clearTimings));
    }
```

File: dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        return sb.toString();
+    }
+
+    public void logTimings() {
+        this.printTimings(true, logger::info);
+    }
+
+    public void logTimings(boolean clearTimings) {
+        printTimings(clearTimings, logger::info);
```
Comment (WalshyDev) [inline dough-common/src/main/java/io/github/bakedlibs/dough/common/DoughTimings.java:66]:
```suggestion
        this.logTimings(true, logger::info);
    }

    public void logTimings(boolean clearTimings) {
        this.logTimings(clearTimings, logger::info);
```

**Comment by WalshyDev** [comment]
Ugh bit of a mess there but there we go, renamed all to logTimings for general consistency. Felt weird to have different names for no real reason. I can understand the player one having a different name but for a single method when 3 are the same, it doesn't feel right.

**Review by WalshyDev — APPROVED** [review]
I'll wait for another reviewer as well but lgtm

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**